app/src/main/java/com/pajato/android/gamechat/exp/CheckersFragment.java

```diff
@@ -422,33 +426,43 @@ private ImageButton makeBoardButton(int index, int sideSize, Map<String, String>
         currentTile.setLayoutParams(param);
         String buttonTag = String.valueOf(index);
         currentTile.setTag(buttonTag);
+        float sp = sideSize / getResources().getDisplayMetrics().scaledDensity;
+        currentTile.setTextSize(COMPLEX_UNIT_SP, (float)(sp * 0.8));
+        currentTile.setTypeface(null, Typeface.BOLD);
+        currentTile.setGravity(Gravity.CENTER);
 
         // Handle the checkerboard positions.
         boolean isEven = index % 2 == 0;
 
         // Create the checkerboard pattern on the button backgrounds.
         if (isEvenEvenOrOddOdd(index, isEven)) {
             currentTile.setBackgroundColor(ContextCompat.getColor(getContext(), android.R.color.white));
-            currentTile.setImageResource(0);
+            currentTile.setText(" ");
         } else {
-            currentTile.setBackgroundColor(ContextCompat.getColor(getContext(), android.R.color.darker_gray));
-            currentTile.setImageResource(0);
+            currentTile.setBackgroundColor(ContextCompat.getColor(getContext(), R.color.colorLightGray));
+            currentTile.setText(" ");
         }
 
         // If the tile is meant to contain a board piece at the start of play, give it a piece.
         if (containsSecondaryPiece(index, isEven) || (pieceType.equals(SECONDARY_PIECE) || pieceType.equals(SECONDARY_KING))) {
```

Comment:
If the database has a board map that maps a location to a piece <String, String> then setting up the board becomes something like: `for (String position : map) {TextView piece = getPiece(position); piece.setText(position, map);}` will take care of painting a board nicely.  This is one of the immense powers that enums bring to the table and why I am so high on them.